File: python/2591.py
class Solution:
    def distMoney(self, money: int, children: int) -> int:
        if money > children * 8:
            return children - 1
        if money == children * 8:
            return children
        if money < children:
            return -1
        if money == 4 and children == 1:
            return -1
        money -= children
        ans = money // 7
        if money % 7 == 3 and ans == children - 1:
            ans -= 1
        return ans

File: go/2591.go
package main

import "fmt"

func distMoney(money int, children int) (ans int) {
	if money > 8*children {
		return children - 1
	}
	if money == 8*children {
		return children
	}
	if money < children {
		return -1
	}
	if money == 4 && children == 1 {
		return -1
	}
	money -= children
	ans = money / 7
	if (ans == children-1) && (money%7 == 3) {
		ans--
	}
	return
}

func main() {
	fmt.Printf("distMoney(20, 3): %v\n", distMoney(20, 3))
}
